python/ray/air/execution/resources/request.py

```diff
@@ -0,0 +1,251 @@
+import json
+from copy import deepcopy
+from inspect import signature
+from typing import List, Type, Dict, Union
+
+from dataclasses import dataclass
+
+from ray.util import placement_group, PublicAPI
+from ray.util.annotations import DeveloperAPI
+
+
+def _sum_bundles(bundles: List[Dict[str, float]]) -> Dict[str, float]:
+    """Sum all resources in a list of resource bundles.
+
+    Args:
+        bundles: List of resource bundles.
+
+    Returns: Dict containing all resources summed up.
+    """
+    resources = {}
+    for bundle in bundles:
+        for k, v in bundle.items():
+            resources[k] = resources.get(k, 0) + v
+    return resources
+
+
+@PublicAPI(stability="beta")
+class ResourceRequest:
+    """Request for resources.
+
+    This class is used internally to define a resource request.
+
+    The resource request must be compatible with the placement group API.
+
+    At a minimum, this will hold at least one bundle specifying the
+    resource requirements for each trial:
```

Comment:
This seems too Tune specific.

Reply:
This is a struct that's both Tune user facing as well as the protocol for ResourceManager (which is supposed to be generic and used across different libraries). How do we best describe it? The ideal is to decouple Tune concept from generic execution concept.